### PR TITLE
Update InfoResponse

### LINKED ISSUE #1066: Add plugins field to InfoResponse
## Description

Add plugins field to InfoResponse

## Motivation

It's here https://editor.swagger.io/?url=https://raw.githubusercontent.com/iotaledger/tips/stardust-api/tips/TIP-0025/core-rest-api.yaml

## Are you planning to do it yourself in a pull request?

No.

### LINKED ISSUE #1091: Update API info response

### PR DESCRIPTION
**Pull request by samuel-rufi**

### CODE REVIEW

File: bee-api/bee-rest-api/src/endpoints/routes/api/v1/info.rs [hunk 1]
```diff
@@ -73,23 +76,35 @@ pub(crate) async fn info<B: StorageBackend>(
...
+            latest_milestone_index: *latest_milestone_index,
+            confirmed_milestone_index: *tangle.get_confirmed_milestone_index(),
+            pruning_index: *tangle.get_pruning_index(),
+        },
+        protocol: ProtocolResponse {
+            network_name: network_id.0,
+            bech32_hrp,
+            min_pow_score: protocol_config.minimum_pow_score(),
+            rent_structure: RentStructureResponse {
+                vbyte_cost: 0,     // TODO
```
Comment (Thoralf-M) [inline bee-api/bee-rest-api/src/endpoints/routes/api/v1/info.rs]:
```suggestion
                vbyte_cost: 500,     // TODO
```
default value here https://github.com/Wollac/protocol-rfcs/blob/protocol-parameters/tips/TIP-0022/tip-0022.md

Comment (Thoralf-M) [inline bee-api/bee-rest-api/src/endpoints/routes/api/v1/info.rs]:
```suggestion
                vbyte_cost: 500,     // TODO
```
default value here https://github.com/Wollac/protocol-rfcs/blob/protocol-parameters/tips/TIP-0022/tip-0022.md

Comment (samuel-rufi) [reply bee-api/bee-rest-api/src/endpoints/routes/api/v1/info.rs]: Oh makes sense, was using these here https://github.com/gohornet/hornet/blob/764e4ee25995b1557129277a97dea244b923513e/core/protocfg/params.go#L39-L41

File: bee-api/bee-rest-api/src/types/responses.rs [hunk 2]
```diff
@@ -14,33 +14,69 @@ use serde::{Deserialize, Serialize};
...
+    pub rent_structure: RentStructureResponse,
+}
+
+/// Returned in InfoResponse.
+/// Rent information about the node.
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct RentStructureResponse {
+    #[serde(rename = "vByteCost")]
+    pub vbyte_cost: u64,
+    #[serde(rename = "vBFactorData")]
```
Comment (Thoralf-M) [inline bee-api/bee-rest-api/src/types/responses.rs]:
```suggestion
    #[serde(rename = "vByteFactorData")]
```

File: bee-api/bee-rest-api/src/types/responses.rs [hunk 3]
```diff
@@ -14,33 +14,69 @@ use serde::{Deserialize, Serialize};
...
+    pub vbyte_factor_data: u64,
+    #[serde(rename = "vBFactorKey")]
```
Comment (Thoralf-M) [inline bee-api/bee-rest-api/src/types/responses.rs]:
```suggestion
    #[serde(rename = "vByteFactorKey")]
```

**Review by Thoralf-M — DISMISSED** [review]

**Review by Thoralf-M — DISMISSED** [review]

File: bee-api/bee-rest-api/src/types/responses.rs [hunk 4]
```diff
@@ -14,33 +14,69 @@ use serde::{Deserialize, Serialize};
...
+    #[serde(rename = "rentStructure")]
+    pub rent_structure: RentStructureResponse,
+}
+
+/// Returned in InfoResponse.
+/// Rent information about the node.
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct RentStructureResponse {
+    #[serde(rename = "vByteCost")]
+    pub vbyte_cost: u64,
```
Comment (grtlr) [inline bee-api/bee-rest-api/src/types/responses.rs:61]: If I'm not mistaken, `vbyte_cost` can be a floating point value.

**Review by grtlr — CHANGES_REQUESTED** [review]

File: bee-api/bee-rest-api/src/types/responses.rs (hunk 4 above)
Comment (samuel-rufi) [reply bee-api/bee-rest-api/src/types/responses.rs:61]: Hmm here it is u64 https://github.com/gohornet/hornet/blob/764e4ee25995b1557129277a97dea244b923513e/core/protocfg/core.go#L61

And in the RFC I couldn't find the datatype https://github.com/muXxer/protocol-rfcs/blob/master/tips/TIP-0019/tip-0019.md

Are there any other resources?

Comment (grtlr) [reply bee-api/bee-rest-api/src/types/responses.rs:61]: Hmm, they might have changed the TIPs again. In [this section](https://github.com/muXxer/protocol-rfcs/blob/master/tips/TIP-0019/tip-0019.md#weights-for-different-outputs) of the TIP they specify floating point values for the weights. Also, the equation for the byte cost specifies rounding down, so at least one of these parameters must be a float.

Comment (Thoralf-M) [reply bee-api/bee-rest-api/src/types/responses.rs:61]: https://github.com/Wollac/protocol-rfcs/blob/protocol-parameters/tips/TIP-0022/tip-0022.md here it's just `500`

Comment (samuel-rufi) [reply bee-api/bee-rest-api/src/types/responses.rs:61]: yes the question is which one. it makes more sense for the `byte_cost` to be a float, but doesn't neccessarly need to be the case I guess. I will ask the Hornet team

Comment (grtlr) [reply bee-api/bee-rest-api/src/types/responses.rs:61]: [This section](https://github.com/muXxer/protocol-rfcs/blob/master/tips/TIP-0019/tip-0019.md#weights-for-different-outputs) shows floating point values for the weights. Other than that, I agree there is not much information about this in the TIP. I've left a comment: https://github.com/iotaledger/tips/pull/39#discussion_r801416125

Comment (samuel-rufi) [reply bee-api/bee-rest-api/src/types/responses.rs:61]: Thanks I have seen - I guess we can let the PR unchanged as both values are integers.

**Review by Thoralf-M — DISMISSED** [review]

**Review by grtlr — DISMISSED** [review]

File: bee-api/bee-rest-api/src/types/responses.rs [hunk 5]
```diff
@@ -14,33 +14,69 @@ use serde::{Deserialize, Serialize};
...
+    pub latest_milestone_timestamp: u64,
+    #[serde(rename = "latestMilestoneIndex")]
+    pub latest_milestone_index: u32,
+    #[serde(rename = "confirmedMilestoneIndex")]
+    pub confirmed_milestone_index: u32,
+    #[serde(rename = "pruningIndex")]
+    pub pruning_index: u32,
+}
+
+/// Returned in InfoResponse.
```
Comment (Adam-Gleave) [inline bee-api/bee-rest-api/src/types/responses.rs]: Maybe:
```suggestion
/// Returned in [`InfoResponse`].
```

File: bee-api/bee-rest-api/src/types/responses.rs [hunk 6]
```diff
@@ -14,33 +14,69 @@ use serde::{Deserialize, Serialize};
...
+    pub status: StatusResponse,
+    pub protocol: ProtocolResponse,
+    pub metrics: MetricsResponse,
+    pub features: Vec<String>,
+    pub plugins: Vec<String>,
+}
+
+impl BodyInner for InfoResponse {}
+
+/// Returned in InfoResponse.
```
Comment (Adam-Gleave) [inline bee-api/bee-rest-api/src/types/responses.rs]:
```suggestion
/// Returned in [`InfoResponse`].
```

File: bee-api/bee-rest-api/src/types/responses.rs [hunk 7]
```diff
@@ -14,33 +14,69 @@ use serde::{Deserialize, Serialize};
...
+    pub network_name: String,
     #[serde(rename = "bech32HRP")]
     pub bech32_hrp: String,
     #[serde(rename = "minPoWScore")]
     pub min_pow_score: f64,
+    #[serde(rename = "rentStructure")]
+    pub rent_structure: RentStructureResponse,
+}
+
+/// Returned in InfoResponse.
```
Comment (Adam-Gleave) [inline bee-api/bee-rest-api/src/types/responses.rs]:
```suggestion
/// Returned in [`InfoResponse`].
```

File: bee-api/bee-rest-api/src/types/responses.rs [hunk 8]
```diff
@@ -14,33 +14,69 @@ use serde::{Deserialize, Serialize};
...
+pub struct RentStructureResponse {
+    #[serde(rename = "vByteCost")]
+    pub vbyte_cost: u64,
+    #[serde(rename = "vByteFactorData")]
+    pub vbyte_factor_data: u64,
+    #[serde(rename = "vByteFactorKey")]
+    pub vbyte_factor_key: u64,
+}
+
+/// Returned in InfoResponse.
```
Comment (Adam-Gleave) [inline bee-api/bee-rest-api/src/types/responses.rs]:
```suggestion
/// Returned in [`InfoResponse`].
```

**Review by Adam-Gleave — COMMENTED** [review]
These ones too :trollface:

**Review by grtlr — APPROVED** [review]

**Review by Thoralf-M — APPROVED** [review]

**Review by Adam-Gleave — APPROVED** [review]

**Review by thibault-martinez — APPROVED** [review]

**Pull request merged**